jsk_pcl_ros/scripts/detect_graspable_poses_pcabase.py

```diff
@@ -0,0 +1,157 @@
+                                               (points[:, 2] > grasp_xy[1] - search_range_m) &
+                                               (points[:, 2] < grasp_xy[1] + search_range_m))]
+                points_depth = points_depth[:, 0]
+            else:
+                rospy.logwarn("direction should be x or z")
+            if len(points_depth) > 0:
+                grasp_position = np.append(grasp_xy, np.mean(points_depth))
+            else:
+                rospy.logwarn("it happens that something wrong")
+            grasp_position_list.append(list(grasp_position))
+
+        pub_msg = PoseArray()
+        pub_msg.header = point_cloud.header
+
+        trans_matrix = tf.transformations.identity_matrix()
+        if self.direction == 'z':
+            trans_matrix[0, 0] = 0
+            trans_matrix[1, 0] = 0
+            trans_matrix[2, 0] = -1
+            trans_matrix[0, 1] = -1 * pca.components_[0, 1]
+            trans_matrix[1, 1] = pca.components_[0, 0]
+            trans_matrix[0, 2] = pca.components_[0, 0]
+            trans_matrix[1, 2] = pca.components_[0, 1]
+            trans_matrix[2, 2] = 0
+
+        elif self.direction == 'x':
+            # this if else is set so that z value of grasp poses' y axies become positive.
+            if pca.components_[0, 0] > 0:
+                trans_matrix[1, 1] = pca.components_[0, 0]
+                trans_matrix[2, 1] = pca.components_[0, 1]
+                trans_matrix[1, 2] = -1 * pca.components_[0, 1]
+                trans_matrix[2, 2] = pca.components_[0, 0]
+            else:
+                trans_matrix[1, 1] = -1 * pca.components_[0, 0]
+                trans_matrix[2, 1] = -1 * pca.components_[0, 1]
+                trans_matrix[1, 2] = pca.components_[0, 1]
+                trans_matrix[2, 2] = -1 * pca.components_[0, 0]
+        else:
+            rospy.logwarn("direction should be x or z")
+
+        quaternion = tf.transformations.quaternion_from_matrix(trans_matrix)
+
+        for grasp_position in grasp_position_list:
+            pose = Pose()
+            if self.direction == 'z':
+                pose.position.x = grasp_position[0]
+                pose.position.y = grasp_position[1]
+                pose.position.z = grasp_position[2]
+            elif self.direction == 'x':
+                pose.position.x = grasp_position[2]
+                pose.position.y = grasp_position[0]
+                pose.position.z = grasp_position[1]
+            else:
+                rospy.logwarn("direction should x or z")
+            pose.orientation.x = quaternion[0]
+            pose.orientation.y = quaternion[1]
+            pose.orientation.z = quaternion[2]
+            pose.orientation.w = quaternion[3]
+            pub_msg.poses.append(pose)
+
+        self.pub_target_poses.publish(pub_msg)
+
+def main():
+    try:
+        DetectGraspablePosesPcabase()
+    except rospy.ROSInterruptException:
```

Comment:
Is this try-except block necessary?
We usually don't have such block in jsk_perception.
https://github.com/jsk-ros-pkg/jsk_recognition/tree/master/jsk_perception/node_scripts